### DISCUSSION TITLE
Obtaining pure HTML code upon publishing for database storage

### DISCUSSION THREAD

**Discussion by bojan27** [puckeditor/puck#272] [Q&A] [answered]
Hi,

how can I obtain the pure HTML code when publishing content. My goal is to store also HTML code in the database for potential use with external services.

Is there a built in function or should I use config?

responsePublishData.content.forEach((contentItem) => {
    const componentRender = config.components[contentItem.type].render;
    const renderedHTML = componentRender(contentItem.props);

    console.log(renderedHTML);
});

**Comment by dohomi** [comment]
I worked on a GrapesJS project recently and in retrospective Puck seems a better fit as my application was written in React and I had to do a lot of tango to make it work with my design.

I would be interested in how this would be archived with Puck, I also would like to elaborate how to convert the content to PDF

**Comment by Rafli-Dewanto** [comment, marked as answer]
if you're using React you can do this

```tsx
const html = renderToStaticMarkup(<Render config={config} data={data} />);
```
